pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -1119,13 +1160,11 @@ public PinotDataType getSingleValueType() {
   }
 
   public static PinotDataType getSingleValueType(Class<?> cls) {
-    PinotDataType pdt = SINGLE_VALUE_TYPE_MAP.get(cls);
-    return (pdt != null) ? pdt : OBJECT;
+    return null == cls ? OBJECT : SINGLE_VALUE_TYPE_MAPPING.apply(cls);
```

Comment:
The input can never be `null`, and we can remove the null check here

Reply:
Actually, null handling is explicitly tested for, which is why the check was added. However, with the latest change (and I suggest we take it or leave it at this point given the limited impact of the change) this is unnecessary.